### PR TITLE
chore: use node 18

### PR DESCRIPTION
**Pull request by jjtang1985**
Note: the tsconfig uses the following:
```
"target": "es2021",
```

Closes SAP/cloud-sdk-backlog#ISSUENUMBER.

- [x] I know which base branch I chose for this PR, as the default branch is `v2-main` now, which is not for v3 development.
- [x] If my change will be merged into the `main` branch (for v3), I've updated (V3-Upgrade-Guide.md)[./V3-Upgrade-Guide.md] in case my change has any implications for users updating to SDK v3

### CODE REVIEW

**Review by fwilhe — CHANGES_REQUESTED** [review]
this means we won't test v2 on non-node18 versions. is that intended? why not do [this](https://github.com/SAP/cloud-sdk-js/pull/3159) instead?

File: test-packages/test-services-e2e/v4/test-service/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es2019",
+    "target": "es2021",
```
Comment (FrankEssenberger) [inline test-packages/test-services-e2e/v4/test-service/tsconfig.json:3]: [q] If I look at the [specs](https://www.typescriptlang.org/tsconfig#target) there are `es2022`and `esnext` as even newer standards. Is there a reason why we use 2021?

**Review by FrankEssenberger — APPROVED** [review]
LGTM only renaming and warning fixes as far as I can see. One question.

File: test-packages/test-services-e2e/v4/test-service/tsconfig.json (hunk 1 above)
Comment (jjtang1985) [reply test-packages/test-services-e2e/v4/test-service/tsconfig.json:3]: With es2022, it introduces lots of compile time/runtime changes and the clients are generated differently.
I switched back to es2021, and now everything are solved right away.
Therefore, I would make this as a first step and merge it.
Based on this, I'll continue my investigation.

**Pull request merged**